source/MapDetailPanel.cpp

```diff
@@ -271,6 +274,35 @@ bool MapDetailPanel::Click(int x, int y, int clicks)
 
 
 
+bool MapDetailPanel::RClick(int x, int y)
+{
+	if(!selectedSystem || !Preferences::Has("System map sends move orders"))
+		return true;
+	// TODO: rewrite the map panels to be driven from interfaces.txt so these XY
+	// positions aren't hard-coded.
+	else if(x >= Screen::Right() - 240 && y >= Screen::Top() + 10 && y <= Screen::Top() + 270)
+	{
+		// Only handle clicks on the actual orbits element, rather than the whole UI region.
+		// (Note: this isn't perfect, and the clickable area extends into the angled sides a bit.)
+		const Point orbitCenter(Screen::TopRight() + Point(-120., 160.));
+		auto uiClick = Point(x, y) - orbitCenter;
+		if(uiClick.Length() > 130)
+			return true;
+		
+		// Only issue movement orders if the player is in-flight.
+		if(player.GetPlanet())
+			GetUI()->Push(new Dialog("You cannot issue fleet movement orders while docked."));
+		else if(!player.HasVisited(selectedSystem))
+			GetUI()->Push(new Dialog("You must visit this system before you can send your fleet there."));
+		else
+			player.SetEscortDestination(selectedSystem, uiClick / scale);
```

Comment:
It's a pity that this goes through the playerInfo class, but I guess that PlayerInfo is the only common shared information between the MapDetailPanel and the engine. Maybe add a comment to explain why the orders don't go to the Engine/AI directly?

Reply:
Yep, the only way to reach AI currently is to go through its owner, Engine, which is owned by the main in-flight panel (as opposed to the map panels).

An opportunity for future architectural changes is to abstract some of the large AI class into e.g. `ShipAI` classes that are owned by any given ship (or group of ships). One would then be able to update a ship's goals, like a desired destination system, without needing the orchestrating AI class.

Reply:
Sounds good. But let's keep this for a future PR.